bip-XXXX.mediawiki

```diff
@@ -0,0 +1,65 @@
+
+==Deployment==
+
+Deployment shall be controlled by hash-power supermajority vote (similar to the technique used in BIP34), but the earliest possible activation time is TBD.
+
+Activation is achieved when 900 of 1,000 consecutive blocks in the best chain have a version number with the first, second, third, fifth, sixth, and thirtieth bits set (0x20000037 in hex). The activation time will be the timestamp of the 900'th block plus a two week (1,209,600 second) grace period to give any remaining miners or services time to upgrade.
+
+At the end of the activation period, which will last for 6 months, there will be a mandatory activation flag day similar to BIP 148.
+
+==Test network==
+
+Test network parameters are the same as the main network, except starting earlier with easier supermajority conditions and a shorter grace period:
+
+    starting time: TBD
+    activation condition: 90 of 100 blocks
+    grace period: 24 hours
+    mandatory activation flag date: TBD
+
+==Rationale==
+
+Time-stamping, or notarization, is an important use of Bitcoin and, importantly, it is a direct use which is essential for a good money. Optimizing the way Bitcoin stores notarization data is important. It is also important to enforce rules that require users to coordinate in a reasonable manner for the good of the community. This will lower transaction fees for all users and even lower transaction fees for users who wish to timestamp their own data. Miners will be incentivized by layer two protocol rewards. See [<a href="#references">3</a>] for further discussion on the rationale.
+
+==Objections to this proposal==
+
+The main objection to this proposal will be the risk and effort required to execute a soft fork. It is important to note that other soft fork proposals may be incorporated into this proposal or this proposal may be incorporated into other proposals which mitigates this risk if the value of all the proposed fork features exceeds the costs. An additional objection may be that current applications make use of the OP_RETURN field and upgrading them will add cost. While this is true, the design of those applications was not optimal and this upgrade is needed. In the long run, it will reduce the running costs of those applications and optimize the use of block-space.
+
+==Compatibility==
+
+As a soft fork, older software will continue to operate without modification, but miners will need to upgrade before the soft fork activates. Additionally users using the OP_RETURN opcode will likely need to update their software to make use of a layer two standard instead of using OP_RETURN directly.
```

Comment:
IMO this should really talk about interaction with BIPs 47, 141, and ideally 300, 301, and other such hypothetical possible future improvements.